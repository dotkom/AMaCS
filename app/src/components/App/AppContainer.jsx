import React, { PropTypes } from "react";
import { Link, Route, Switch } from "react-router-dom";

import Application from "./Application"
import Heading from "../Heading";
import Home from "./Home"
import NotFound from "../NotFound"
import 'assets/css/base.css';

class AppContainer extends React.Component {
  constructor(props) {
    super(props)

    this.state = {

    };
  }

  render() {
    const { match, user } = this.props
    return(
      <div>
        <Heading />
        <main>
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={`/application`} render={props =>
              <Application
                user={user}
                {...props}
              />
            } />
            <Route component={NotFound} />
          </Switch>
        </main>
      </div>
    );
  }
}

export default AppContainer;
