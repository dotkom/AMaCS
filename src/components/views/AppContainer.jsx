import React from 'react';
import { Route, Switch } from 'react-router-dom';

import Application from 'components/views/commitee/Application';
import AuthenticationEndpoint from 'components/views/AuthenticationEndpoint';
import Heading from 'components/misc/Heading';
import Home from 'components/views/commitee/Home';
import NotFound from 'components/views/NotFound';
import Thankyou from 'components/views/Thankyou';
import _s from 'assets/css/base.module.scss';

class AppContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const { user } = this.props;
    return (
      <div>
        <Heading />
        <main>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/application" render={(props) => <Application user={user} {...props} />} />
            <Route exact path="/thankyou" component={Thankyou} />
            <Route exact path="/auth" component={AuthenticationEndpoint} />
            <Route component={NotFound} />
          </Switch>
        </main>
        <div className={_s.footer}>
          <p>
            Dersom du opplever problemer eller finner feil, ta kontakt med{' '}
            <a href="mailto:dotkom@online.ntnu.no">dotkom@online.ntnu.no</a>.
          </p>
          <p>
            Får du ikke sendt inn søknaden, eller vil du heller bruke e-post? Send søknaden din til{' '}
            <a href="mailto:opptak@online.ntnu.no">opptak@online.ntnu.no</a>.
          </p>
        </div>
      </div>
    );
  }
}

export default AppContainer;
