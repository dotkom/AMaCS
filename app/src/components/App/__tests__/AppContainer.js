import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';

import AppContainer from '../AppContainer';

const createAppContainer = (path, user) => (
  <AppContainer
    match={{ path }}
    user={user}
  />
)

describe('AppContainer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      createAppContainer('/')
    );
    expect(wrapper).toMatchSnapshot();
  });
});
