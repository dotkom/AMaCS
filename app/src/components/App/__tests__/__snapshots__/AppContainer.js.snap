// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders correctly 1`] = `
<div>
  <Heading />
  <main>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        exact={true}
        path="/application"
        render={[Function]}
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
</div>
`;
